host/xtest/ffa_spmc_1000.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2022, Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <fcntl.h>
+#include "include/uapi/linux/arm_ffa_user.h"
+#include "include/ffa.h"
+#include <stdio.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+#include "xtest_helpers.h"
+#include "xtest_test.h"
+
+#define FFA_DRIVER_FS_PATH	"/sys/kernel/debug/arm_ffa_user"
+#define SPMC_TEST_OK 0xaa
+#define exit_test goto out;
+
+enum sp_tests {
+	EP_TEST_SP,
+	EP_TEST_SP_COMMUNICATION,
+};
+
+static int ffa_fd = -1;
+
+const char test_endpoint1_uuid[] = "5c9edbc3-7b3a-4367-9f83-7c191ae86a37";
+const char test_endpoint2_uuid[] = "7817164c-c40c-4d1a-867a-9bb2278cf41a";
+const char test_endpoint3_uuid[] = "23eb0100-e32a-4497-9052-2f11e584afa6";
+
+struct ffa_ioctl_ep_desc test_endpoint1 = {
+		.uuid_ptr = (uint64_t) test_endpoint1_uuid,
+		.id = 0
```

Comment:
Personally I wouldn't set `.id` since it is guaranteed to be zero

Reply:
Ack